crates/pallets/starknet/src/tests/l1_handler_validation.rs

```diff
@@ -40,7 +40,7 @@ fn should_accept_unused_nonce() {
 
         let tx = UserAndL1HandlerTransaction::L1Handler(transaction, Fee(100));
 
-        assert_matches!(Starknet::validate_unsigned_tx_nonce(&tx), Ok(TxPriorityInfo::L1Handler));
+        assert_matches!(Starknet::validate_unsigned_tx_nonce(&tx), Ok(TxPriorityInfo::L1Handler { nonce: _ }));
```

Comment:
You can check that the nonce is correct too.
You specified `let nonce: u64 = 1;`